src/effects/builtin/whitenoiseeffect.cpp

```diff
@@ -0,0 +1,63 @@
+#include "whitenoiseeffect.h"
+
+#include <QtDebug>
+#include <QRandomGenerator>
+
+// static
+QString WhiteNoiseEffect::getId() {
+    return "org.mixxx.effects.noise";
+}
+
+// static
+EffectManifestPointer WhiteNoiseEffect::getManifest() {
+    EffectManifestPointer pManifest(new EffectManifest());
+    pManifest->setId(getId());
+    pManifest->setName(QObject::tr("White Noise"));
+    pManifest->setAuthor("The Mixxx Team");
+    pManifest->setVersion("1.0");
+    pManifest->setDescription(QObject::tr("Creates active white noise sound"));
+
+    // This is loudness parameter
+    EffectManifestParameterPointer intensity = pManifest->addParameter();
+    intensity->setId("Loudness");
+    intensity->setName(QObject::tr("Loudness"));
+    intensity->setDescription(QObject::tr("Set the loudness which makes the noise louder or more quiet"));
+    intensity->setControlHint(EffectManifestParameter::ControlHint::KNOB_LOGARITHMIC);
+    intensity->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    intensity->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    intensity->setDefaultLinkType(EffectManifestParameter::LinkType::LINKED);
+    intensity->setMinimum(0);
+    intensity->setDefault(0);
+    intensity->setMaximum(1);
+
+    return pManifest;
+}
+
+WhiteNoiseEffect::WhiteNoiseEffect(EngineEffect* pEffect)
+        : m_pLoudnessParameter(pEffect->getParameterById("Loudness")) {
+}
+
+WhiteNoiseEffect::~WhiteNoiseEffect() {
+}
+
+void WhiteNoiseEffect::processChannel(
+        const ChannelHandle& handle,
+        NoiseGroupState* pGroupState,
+        const CSAMPLE* pInput,
+        CSAMPLE* pOutput,
+        const mixxx::EngineParameters& bufferParameters,
+        const EffectEnableState enableState,
+        const GroupFeatureState& groupFeatures) {
+    Q_UNUSED(handle);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(enableState);
+    Q_UNUSED(pGroupState);
+    Q_UNUSED(pInput);
+
+    CSAMPLE Loudness = m_pLoudnessParameter->value();
+
+    for (unsigned int i = 0; i < bufferParameters.samplesPerBuffer(); i ++) {
+        float output = static_cast<float>(QRandomGenerator::global()->generateDouble() * Loudness);
```

Comment:
I think this should be a dry/wet parameter. By not including the input signal at all, the only way to use this as part of a chain is at the beginning of the chain.

Also, this should be a ramping value. Otherwise, there are stairstep jumps in the volume between cycles of the audio thread instead of continuous changes. You can look at the echo effect for an example of how to implement this.

Reply:
I have no idea how to build what you said and maybe some assistance would be good. I saw that the echo effect has 

```
RampingValue<CSAMPLE_GAIN> send(send_current, gs.prev_send,
                                    bufferParameters.framesPerBuffer());
```

For making the ramping value and

```
CSAMPLE_GAIN send_ramped = send.getNext();
```

In the for loop

Reply:
Yes, you need something like that. You need to store the value of the parameter in the GroupState. That is what `gs.prev_send` is for there.